code/modules/reagents/chemistry/holder/reactions.dm

```diff
@@ -65,7 +65,7 @@
 					if(reaction.required_container)
 						if(reaction.required_container_accepts_subtypes && !istype(cached_my_atom, reaction.required_container))
 							continue
-						else if(cached_my_atom.type != reaction.required_container)
+						else if((cached_my_atom.type != reaction.required_container) && !reaction.required_container_accepts_subtypes)
```

Comment:
We already checked it once above, we should just re-arrange this
```dm
if(reaction.required_container)
    if(reaction.required_container_accepts_subtypes)
        if(istype(cached_my_atom, reaction.required_container))
            continue
    else 
        if(cached_my_atom.type != reaction.required_container)
            continue
```

Reply:
Correction its `!istype()`. You forgot the !